### PR TITLE
[BP-1.14][FLINK-27399][Connector/Pulsar] [Connector/Pulsar] Change initial consuming position setting logic for better handle the checkpoint.

### PR DESCRIPTION
**Pull request by syhily**
This is a backport PR for https://github.com/apache/flink/pull/19972.

### CODE REVIEW

File: flink-connectors/flink-connector-pulsar/pom.xml [hunk 1]
```diff
@@ -36,13 +36,15 @@ under the License.
 	<packaging>jar</packaging>
 
 	<properties>
-		<pulsar.version>2.8.0</pulsar.version>
+		<pulsar.version>2.9.1</pulsar.version>
```
Comment (tisonkun) [inline flink-connectors/flink-connector-pulsar/pom.xml:39]: How is this version bump necessary?

Basically, we should not bump version in a patch version, especially which can introduce new features.

**Review by tisonkun — COMMENTED** [review]
Thanks for cherry-picking this patch. One comment is inline.

Also, I'd like to confirm that you bring the test utils from master to 1.14 for Pulsar Connector tests. Is it right?

File: flink-connectors/flink-connector-pulsar/pom.xml (hunk 1 above)
Comment (syhily) [reply flink-connectors/flink-connector-pulsar/pom.xml:39]: We have to use this version bump because the underlying API is introduced on Pulsar 2.9.x.

**Comment by syhily** [comment]
> Thanks for cherry-picking this patch. One comment is inline.
> 
> Also, I'd like to confirm that you bring the test utils from master to 1.14 for Pulsar Connector tests. Is it right?

Yep. The testing tools are introduced for adding transaction support for mocked Pulsar.

**Comment by syhily** [comment]
@PatrickRen Can you help me review this backport issue?

**Review by tisonkun — APPROVED** [review]
LGTM. Merging...

Anyone can leave a review comment after merge.

**Pull request merged**